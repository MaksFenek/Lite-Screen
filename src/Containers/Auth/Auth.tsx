// React
import React from 'react';
import { Link } from 'react-router-dom';

// Components
import AuthForm from './AuthForm/AuthForm';
import Navbar from '../Generic/Navbar/Navbar';

// Images
import LoginImg from '../../Icons/undraw_authentication_fsn5.svg';
import SignupImg from '../../Icons/undraw_friendship_mni7.svg';

// Styles and Material ui
import './Auth.scss';
import { Button } from '@material-ui/core';

// ==== TypeScript ====
interface IAuth {
  type: 'login' | 'signup';
}

const Auth: React.FC<IAuth> = ({ type }) => {
  return (
    <>
      <Navbar>
        <Link to={type === 'login' ? `/signup` : `/`}>
          <Button variant='contained' color='primary'>
            {type === 'login' ? `Sign Up` : `Login`}
          </Button>
        </Link>
      </Navbar>
      <section className='auth'>
        <div className='container'>
          <div className='auth-body'>
            {window.innerWidth > 768 && (
              <img
                src={type === 'login' ? LoginImg : SignupImg}
                className={type === 'login' ? 'login-image' : 'signup-image'}
                alt=''
              />
            )}

            <AuthForm type={type === 'login' ? 'login' : 'signup'} />
          </div>
        </div>
      </section>
    </>
  );
};

export default React.memo(Auth);
