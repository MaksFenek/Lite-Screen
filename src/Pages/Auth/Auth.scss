.auth {
  margin: 0 10px;
  padding-top: 90px;
  min-height: calc(100vh - 100px);
  &-body {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  .signup-image {
    height: 410px;
    width: 600px;
  }
  .login-image {
    height: 404px;
    width: 586px;
  }
}

@media screen and (max-width: 767px) {
  .auth {
    .signup-image,
    .login-image {
      display: none;
    }
    &-body {
      justify-content: center;
    }
  }
}

@media screen and (max-width: 900px) {
  .auth {
    .signup-image,
    .login-image {
      margin-bottom: 20px;
    }
    &-body {
      flex-direction: column;
    }
  }
}
