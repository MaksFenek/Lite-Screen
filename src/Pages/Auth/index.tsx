import React from 'react';

import AuthForm from '../../Containers/Reg/AuthForm';

import Navbar from '../../Containers/Generic/Navbar';

import LoginImg from '../../Icons/undraw_authentication_fsn5.svg';
import SignupImg from '../../Icons/undraw_friendship_mni7.svg';

import { Button } from '@material-ui/core';
import { Link } from 'react-router-dom';

interface IAuth {
  type: 'login' | 'signup';
}

const Auth: React.FC<IAuth> = (props) => {
  const type = props.type;

  const width: number = window.innerWidth;

  return (
    <>
      <Navbar>
        <Link to={type === 'login' ? `/signup` : `/`}>
          <Button variant='contained' color='primary'>
            {type === 'login' ? `Sign Up` : `Login`}
          </Button>
        </Link>
      </Navbar>
      <section className='auth'>
        <div className='container'>
          <div className='auth-body'>
            {width < 767 ? (
              ''
            ) : (
              <img
                src={type === 'login' ? LoginImg : SignupImg}
                className={type === 'login' ? 'login-image' : 'signup-image'}
                alt=''
              />
            )}

            <AuthForm type={type === 'login' ? 'login' : 'signup'} />
          </div>
        </div>
      </section>
    </>
  );
};

export default Auth;
