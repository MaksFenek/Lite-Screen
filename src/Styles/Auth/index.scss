.auth {
  margin: 0 10px;
  padding-top: 90px;
  min-height: calc(100vh - 100px);
  &-body {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  .signup-image {
    height: 410px;
    width: 600px;
  }
  .login-image {
    height: 404px;
    width: 586px;
  }

  &-forms {
    padding: 30px 20px;
    background: #bbdefb;
    border: 1px solid #64b5f6;
    border-radius: 10px;
    box-shadow: 2px 3px 6px #e0e0e0;

    h3 {
      font-size: 25px;
      text-align: center;
      line-height: 0;
      color: #616161;
    }
  }

  &-form {
    display: flex;
    flex-direction: column;

    &-input {
      margin: 10px;
    }
    p {
      color: red;
      display: block;
      max-width: 200px;
    }
  }
}

@media screen and (max-width: 767px) {
  .auth {
    .signup-image,
    .login-image {
      display: none;
    }
    &-body {
      justify-content: center;
    }

    &-forms {
      margin: 0;
    }
  }
}

@media screen and (max-width: 900px) {
  .auth {
    .signup-image,
    .login-image {
      margin-bottom: 20px;
    }
    &-body {
      flex-direction: column;
    }
  }
}
