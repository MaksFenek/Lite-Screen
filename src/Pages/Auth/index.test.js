import React from 'react';
import Auth from './index';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('<Auth/>', () => {
  const LoginTree = renderer.create(
    <MemoryRouter>
      <Auth type='login' />
    </MemoryRouter>
  );
  const SignupTree = renderer.create(
    <MemoryRouter>
      <Auth type='signup' />
    </MemoryRouter>
  );

  it('should render correctly', () => {
    expect(LoginTree).toMatchSnapshot();
    expect(SignupTree).toMatchSnapshot();
  });
});
