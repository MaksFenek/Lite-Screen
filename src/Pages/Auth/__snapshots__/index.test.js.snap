// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth/> should render correctly 1`] = `
Array [
  <nav
    className="nav"
  >
    <div
      className="container"
    >
      <div
        className="navbar"
      >
        <div
          className="navbar-logo"
        >
          <svg
            height="48"
            viewBox="0 0 64 64"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Tempered Glass-Smartphone"
              id="Tempered_Glass-Smartphone"
            >
              <rect
                fill="#3d9ae2"
                height="54"
                rx="4"
                width="30"
                x="5"
                y="7"
              />
              <path
                d="m35 11v46h-4a3.995 3.995 0 0 1 -4-4v-46h4a4 4 0 0 1 4 4z"
                fill="#1e81ce"
              />
              <path
                d="m26 7v3a1 1 0 0 1 -1 1h-10a1 1 0 0 1 -1-1v-3z"
                fill="#ff9811"
              />
              <path
                d="m59 7v46a4 4 0 0 1 -4 4h-22a3.995 3.995 0 0 1 -4-4v-46a4 4 0 0 1 4-4h22a4 4 0 0 1 4 4z"
                fill="#5aaae7"
              />
              <path
                d="m50 3v3a1 1 0 0 1 -1 1h-10a1 1 0 0 1 -1-1v-3z"
                fill="#e6e7e8"
              />
              <circle
                cx="44"
                cy="51"
                fill="#e6e7e8"
                r="2"
              />
              <path
                d="m35 40-6 6v7a4.025 4.025 0 0 0 1.17 2.83l4.83-4.83 24-24v-11z"
                fill="#96c8ef"
              />
              <path
                d="m35 34-6 6v6l6-6 24-24v-6z"
                fill="#78b9eb"
              />
              <circle
                cx="20"
                cy="55"
                fill="#ff9811"
                r="2"
              />
            </g>
          </svg>
        </div>
        <div
          className="auth-navbar"
        >
          <a
            href="/signup"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Sign Up
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <section
    className="auth"
  >
    <div
      className="container"
    >
      <div
        className="auth-body"
      >
        <img
          alt=""
          className="login-image"
          src="undraw_authentication_fsn5.svg"
        />
        <div
          className="auth-forms"
        >
          <h3>
            Login
          </h3>
          <form
            className="auth-form"
            noValidate={true}
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline auth-form-input"
              onClick={[Function]}
              required={true}
            >
              <input
                aria-label="Email"
                className="MuiInputBase-input MuiInput-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="E-mail"
                required={true}
                type="text"
              />
            </div>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline auth-form-input"
              onClick={[Function]}
              required={true}
            >
              <input
                aria-label="Password"
                className="MuiInputBase-input MuiInput-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
            </div>
            
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Send
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`<Auth/> should render correctly 2`] = `
Array [
  <nav
    className="nav"
  >
    <div
      className="container"
    >
      <div
        className="navbar"
      >
        <div
          className="navbar-logo"
        >
          <svg
            height="48"
            viewBox="0 0 64 64"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Tempered Glass-Smartphone"
              id="Tempered_Glass-Smartphone"
            >
              <rect
                fill="#3d9ae2"
                height="54"
                rx="4"
                width="30"
                x="5"
                y="7"
              />
              <path
                d="m35 11v46h-4a3.995 3.995 0 0 1 -4-4v-46h4a4 4 0 0 1 4 4z"
                fill="#1e81ce"
              />
              <path
                d="m26 7v3a1 1 0 0 1 -1 1h-10a1 1 0 0 1 -1-1v-3z"
                fill="#ff9811"
              />
              <path
                d="m59 7v46a4 4 0 0 1 -4 4h-22a3.995 3.995 0 0 1 -4-4v-46a4 4 0 0 1 4-4h22a4 4 0 0 1 4 4z"
                fill="#5aaae7"
              />
              <path
                d="m50 3v3a1 1 0 0 1 -1 1h-10a1 1 0 0 1 -1-1v-3z"
                fill="#e6e7e8"
              />
              <circle
                cx="44"
                cy="51"
                fill="#e6e7e8"
                r="2"
              />
              <path
                d="m35 40-6 6v7a4.025 4.025 0 0 0 1.17 2.83l4.83-4.83 24-24v-11z"
                fill="#96c8ef"
              />
              <path
                d="m35 34-6 6v6l6-6 24-24v-6z"
                fill="#78b9eb"
              />
              <circle
                cx="20"
                cy="55"
                fill="#ff9811"
                r="2"
              />
            </g>
          </svg>
        </div>
        <div
          className="auth-navbar"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Login
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <section
    className="auth"
  >
    <div
      className="container"
    >
      <div
        className="auth-body"
      >
        <img
          alt=""
          className="signup-image"
          src="undraw_friendship_mni7.svg"
        />
        <div
          className="auth-forms"
        >
          <h3>
            Sign Up
          </h3>
          <form
            className="auth-form"
            noValidate={true}
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline auth-form-input"
              onClick={[Function]}
              required={true}
            >
              <input
                aria-label="Email"
                className="MuiInputBase-input MuiInput-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="E-mail"
                required={true}
                type="text"
              />
            </div>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline auth-form-input"
              onClick={[Function]}
              required={true}
            >
              <input
                aria-label="Password"
                className="MuiInputBase-input MuiInput-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
            </div>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline auth-form-input"
              onClick={[Function]}
              required={true}
            >
              <input
                aria-label="FirstName"
                className="MuiInputBase-input MuiInput-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="First name"
                required={true}
                type="text"
              />
            </div>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline auth-form-input"
              onClick={[Function]}
              required={true}
            >
              <input
                aria-label="SecondName"
                className="MuiInputBase-input MuiInput-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Second name"
                required={true}
                type="text"
              />
            </div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Send
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </section>,
]
`;
